### DISCUSSION TITLE
Does wagmi enforce checksum addresses now?

### DISCUSSION THREAD

**Discussion by razgraf** [wevm/wagmi#3598] [Q&A]
### Context

While updating to wagmi@2 and viem@2 I've been bumping into an error when preparing a transaction for [`writeContract`](https://wagmi.sh/core/api/actions/writeContract):

`ConnectorAccountNotFoundError: Account "0x..." not found for connector "MetaMask"` 

Connectors (and the config) were being passed down correctly, the one being used at that time included the required account (checked with `connector.getAccounts()`) but the error was still there. 

Digging a bit deeper in the codebase, looking for usage of the error above, I found the `getConnectorClient` method which checks that the passed account is in fact authorized by the connector.

https://github.com/wevm/wagmi/blob/b2d73f8bc7afa21441b6e8f5ecd451d1e069228a/packages/core/src/actions/getConnectorClient.ts#L86-L97

### Problem

As you can see, the checks over connected account are over the **base strings**, without auto-formatting included. What happened to me was that I was passing a **lowercased** address, while the ones retrieved from the connector were all **checksummed**. 

I finally solved it by making sure addresses passed to the write methods are all pre-formatted using viem's [`getAddress`](https://viem.sh/docs/utilities/getAddress).

### Questions

- Does wagmi/viem now ask for fully formatted addresses? Is this documented (and maybe I missed it)
- If not, should the checks mentioned above use a pre-formatting step to have all strings look the same
- This should be true, but do all connectors return addresses in the same format or is there a step where this is checked/enforced to avoid mismatches?

**Comment by tmm** [comment]
Feel free to submit a PR for this specific location. In general, all addresses that Wagmi gives you (e.g. from connectors) should be checksummed. If you notice any other areas that might have issues PRs welcome!

**Reply by razgraf** [reply]
Thanks for the reply! Submitted a fix in #3605.
